src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,135 @@
+                Book nextBook = new Book(id, title, price);
+                acquiredBooks.add(nextBook);
+            }
+            return acquiredBooks;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error during finding all books", e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String deleteRowByIdQuery = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(deleteRowByIdQuery)) {
+            preparedStatement.setLong(FIRST_PARAMETER,id);
+            int affectedRows = preparedStatement.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error deleting book by id: " + id, e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String updateQuery = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(updateQuery)) {
+            preparedStatement.setString(FIRST_PARAMETER, book.getTitle());
+            preparedStatement.setBigDecimal(SECOND_PARAMETER, book.getPrice());
+            preparedStatement.setLong(THIRD_PARAMETER, book.getId());
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows == 0) {
+                throw new RuntimeException("No updated data! Data should to update:"
+                        + book);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error during updating books database with book: "
+                    + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String deleteRowByIdQuery = "SELECT * FROM books WHERE id = ?";
+        Optional<Book> value = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(deleteRowByIdQuery)) {
+            preparedStatement.setLong(FIRST_PARAMETER,id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString(TITLE_COLUMN_LABEL);
+                BigDecimal price = resultSet.getObject(PRICE_COLUMN_LABEL,
+                        BigDecimal.class);
+                Book aquiredBook = new Book(id, title, price);
+                value = Optional.of(aquiredBook);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error during finding book by id: " + id, e);
```

Comment:
Maybe you need to put this logic in a separate method like private Book mapRowToBook(ResultSet resultSet) {
} your code will become more readable. This is my recommendation.

Reply:
Good solution. Let's improve a little bit.